clients/src/main/java/org/apache/kafka/clients/producer/BufferExhaustedException.java

```diff
@@ -16,13 +16,13 @@
  */
 package org.apache.kafka.clients.producer;
 
-import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.errors.TimeoutException;
 
 /**
  * This exception is thrown if the producer is in non-blocking mode and the rate of data production exceeds the rate at
  * which data can be sent for long enough for the allocated buffer to be exhausted.
  */
-public class BufferExhaustedException extends KafkaException {
+public class BufferExhaustedException extends TimeoutException {
```

Comment:
Could you add comment for this inheritance? this change is for keeping compatibility.

Reply:
Done

Reply:
Is this for backwards compatibility? It seems to me that part of the reason is to allow the users to handle all timeout related type of exceptions with one catch clause.

Reply:
> Is this for backwards compatibility?

the client code catching TimeoutException to handle memory issue will be broken if BufferExhaustedException does not extend TimeoutException.

Reply:
It is a bit hen and egg I guess. The main reason for adding it in this case was compatibility (at least to my mind), but what you say is a very welcome side effect.
Happy to add a clarification to the comment of course if we feel this makes sense.